README.md

```diff
@@ -1,2 +1,80 @@
+  result of traversing the whole (or part) of the wizard.
+* **Usage:** Providing the endpoints your system will use to manage each user's navigation through any of the
+  previously defined wizards, collecting data and figuring out what options to display in each step.
+
+### Limitations
+
+The original goal of this project is to serve as an example for [this training session](https://codebeameurope.com/trainings/everything-you-always-wanted-to-know-about-testing-on-the-beam/).
+Therefore, it's not expected to be production-ready by any means. IF there is traction and need, that may change in the
+future. Should that happen, this section will be removed. Before that, **USE IT AT YOUR OWN RISK**.
+
+## Usage
+
+### Configuration
+
+To define a wizard with `rincewind`, you have to provide it with the corresponding configuration.
+You can do that using the `create_wizard/2` function and provided the corresponding `rincewind:wizard()` definition,
+like this:
+
+```erlang
+rincewind:create_wizard(WizardName, #{phases := […]})
+```
+
+[//]: # (TODO: Complete this section with the actual definition of the phases and the rest of the parameters)
+
+### Consumption
+
+Once you have a user that wants to go through a wizard, you can manage their progress using the following functions:
+
+```erlang
+%% Kick off the wizard process and get a reference for it
+{ok, WizardProcess} = rincewind:start(WizardName, UserName),
+%% Get the fields and options for the current phase
+{ok, #{phase := FirstPhaseName, fields := […]}} = rincewind:current_phase(WizardProcess),
+%% Receive input from the user for the current phase (it returns the definition for the next phase, if there is one)
+{next_phase, #{phase := SecondPhaseName, fields := […]}} =
+    rincewind:submit(WizardProcess, #{phase => FirstPhaseName, field_values => […]}),
+%% If the submitted values are deemed invalid, the corresponding validation errors are returned
+{invalid, #{phase := SecondPhaseName, field_errors := […]}} =
+    rincewind:submit(WizardProcess, #{phase => SecondPhaseName, field_values => […]}),
+%% If there are no further phases, the final result is returned
+{done, #{overall_result := OverallResult, phase_results := [#{phase := FirstPhaseName, value := FirstPhaseValue}, …]}} =
+    rincewind:submit(WizardProcess, #{phase => SecondPhaseName, field_values => […]}),
+```
+
+While that's the happy path, Rincewind also provides the means for the user to jump through phases (if that's allowed by
+the wizard definition):
+
+```erlang
+%% Get the full wizard definition (it can be used to navigate through the phases)
+#{name := WizardName, phases := […]} = rincewind:wizard_definition(WizardProcess),
+%% Skip the current phase
+{next_phase, #{phase := SecondPhaseName, fields := […]}} = rincewind:skip_phase(WizardProcess),
+%% Move back to a previous phase
+{next_phase, #{phase := FirstPhaseName, fields := […]}} = rincewind:jump_back(WizardProcess, FirstPhaseName),
```

Comment:
Though it makes sense to yield `{next_phase, ...}` wouldn't it be better to have `{previous_phase, ...}` to have a clear distinction between progressing in the wizard vs. going back and fixing something in a previous state?

Reply:
🤔 Well, but… the new next phase (or well… the **current** phase) in this case _will be_ `FristPhaseName`, right? I mean… this function, like `skip_phase/1` and `submit/2`, will return the next phase that has to be shown to the user. At least, that was my idea.

Reply:
Works for me, no probs 👍🏻